src/transforms/lua/v2/interop/event.rs

```diff
@@ -25,14 +27,29 @@ impl<'a> FromLua<'a> for Event {
                 })
             }
         };
-        match (table.get("log")?, table.get("metric")?) {
-            (LuaValue::Table(log), LuaValue::Nil) => {
-                Ok(Event::Log(LogEvent::from_lua(LuaValue::Table(log), ctx)?))
+        let metadata = match table.get("metadata")? {
+            metadata @ LuaValue::Table(_) => EventMetadata::from_lua(metadata, ctx)?,
+            LuaValue::Nil => EventMetadata::now(),
+            _ => {
+                return Err(LuaError::FromLuaConversionError {
+                    from: type_name(&value),
+                    to: "EventMetadata",
+                    message: Some("EventMetadata should be a table".to_string()),
+                })
             }
-            (LuaValue::Nil, LuaValue::Table(metric)) => Ok(Event::Metric(Metric::from_lua(
-                LuaValue::Table(metric),
-                ctx,
-            )?)),
+        };
+        match (table.get("log")?, table.get("metric")?) {
+            // This is less than ideal. The log or metric below is
+            // created with stub metadata which is then replaced by the
+            // actual metadata. However, I don't know how to properly
+            // pass the metadata parsed above down into these `from_lua`
+            // functions.
```

Comment:
It is now, but it will grow as additional pieces of metadata are added to the structure. Still, most of those will be stubs or other low-cost defaults. So yes, it's not huge, it's just a wart I couldn't figure out how to work around that maybe a reviewer or future coder would have a better idea about.

Reply:
Good point, then it makes sense to leave the comment.